### PR TITLE
chore(EventSubscriber): return a 404 status when route doesnt exist

### PR DESCRIPTION
**Pull request by jeremyjaouen**
## Description

this PR intends to fix an error of status code while the endpoint doesn't exists on APIv2

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [x] 23.04.x (master)

<h2> How this pull request can be tested ? </h2>

Call an non existent endpoint, the HTTP Status code should be 404 and the content

```
{
  "code": 404,
  "message": "No route found for \"GET http://localhost:8080/centreon/api/latest/<endpoint>""
}
```

#### Community contributors & Centreon team

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/EventSubscriber/CentreonEventSubscriber.php [hunk 1]
```diff
@@ -357,11 +358,14 @@ public function onKernelException(ExceptionEvent $event): void
          * If we don't do that a HTML error will appeared.
          */
         if ($errorIsBeforeController) {
-            if ($event->getThrowable()->getCode() !== 403) {
+            if ($event->getThrowable()->getCode() >= 500) {
                 $errorCode = $event->getThrowable()->getCode() > 0
```
Comment (kduret) [inline centreon/src/EventSubscriber/CentreonEventSubscriber.php:362]: `> 0` condition is always true

Comment (jeremyjaouen) [reply centreon/src/EventSubscriber/CentreonEventSubscriber.php:362]: indeed

File: centreon/src/EventSubscriber/CentreonEventSubscriber.php [hunk 2]
```diff
@@ -357,11 +358,12 @@ public function onKernelException(ExceptionEvent $event): void
...
-                    : Response::HTTP_INTERNAL_SERVER_ERROR;
+            if ($event->getThrowable()->getCode() >= 500) {
+                $errorCode = $event->getThrowable()->getCode();
                 $statusCode = Response::HTTP_INTERNAL_SERVER_ERROR;
+            } elseif ($event->getThrowable() instanceof NotFoundHttpException) {
+                $errorCode = Response::HTTP_NOT_FOUND;
+                $statusCode = Response::HTTP_NOT_FOUND;
             } else {
                 $errorCode = $event->getThrowable()->getCode();
                 $statusCode = Response::HTTP_FORBIDDEN;
```
Comment (jeremyjaouen) [inline centreon/src/EventSubscriber/CentreonEventSubscriber.php:369]:
```suggestion
                $errorCode = $event->getThrowable()->getCode();
                $statusCode = $event->getThrowable()->getCode() > 0
                    ? $event->getThrowable()->getCode()
                    : Response::HTTP_INTERNAL_SERVER_ERROR;
```

**Review by kduret — APPROVED** [review]

File: centreon/src/EventSubscriber/CentreonEventSubscriber.php [hunk 3]
```diff
@@ -357,14 +358,17 @@ public function onKernelException(ExceptionEvent $event): void
...
-            if ($event->getThrowable()->getCode() !== 403) {
-                $errorCode = $event->getThrowable()->getCode() > 0
-                    ? $event->getThrowable()->getCode()
-                    : Response::HTTP_INTERNAL_SERVER_ERROR;
+            if ($event->getThrowable()->getCode() >= 500) {
+                $errorCode = $event->getThrowable()->getCode();
                 $statusCode = Response::HTTP_INTERNAL_SERVER_ERROR;
+            } elseif ($event->getThrowable() instanceof NotFoundHttpException) {
+                $errorCode = Response::HTTP_NOT_FOUND;
+                $statusCode = Response::HTTP_NOT_FOUND;
```
Comment (Fxdpt) [inline centreon/src/EventSubscriber/CentreonEventSubscriber.php:366]:
```suggestion
            } elseif ($event->getThrowable()->getCode() === 403) {
                $errorCode = $event->getThrowable()->getCode();
                $statusCode = Response::HTTP_FORBIDDEN;
            } elseif ($event->getThrowable() instanceof NotFoundHttpException) {
                $errorCode = Response::HTTP_NOT_FOUND;
                $statusCode = Response::HTTP_NOT_FOUND;
```

**Review by abuathier — APPROVED** [review]

File: centreon/src/EventSubscriber/CentreonEventSubscriber.php [hunk 4]
```diff
@@ -357,14 +358,20 @@ public function onKernelException(ExceptionEvent $event): void
...
                 $errorCode = $event->getThrowable()->getCode();
                 $statusCode = Response::HTTP_FORBIDDEN;
+            } elseif ($event->getThrowable() instanceof NotFoundHttpException) {
+                $errorCode = Response::HTTP_NOT_FOUND;
+                $statusCode = Response::HTTP_NOT_FOUND;
+            } else {
+                $errorCode = $event->getThrowable()->getCode();
+                $statusCode = $event->getThrowable()->getCode() > 0
+                    ? $event->getThrowable()->getCode()
+                    : Response::HTTP_INTERNAL_SERVER_ERROR;
```
Comment (abuathier) [inline centreon/src/EventSubscriber/CentreonEventSubscriber.php:374]: If the code is not negative, this should be enough.
```suggestion
                $statusCode = $event->getThrowable()->getCode()
                    ?: Response::HTTP_INTERNAL_SERVER_ERROR;
```

**Review by abuathier — DISMISSED** [review]

**Review by dmyios — DISMISSED** [review]

**Review by dmyios — DISMISSED** [review]

**Review by abuathier — APPROVED** [review]

**Review by abuathier — APPROVED** [review]

**Review by dmyios — APPROVED** [review]

**Pull request merged**